From 29b37e45577c0921846c1709a190f08a3b032666 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Fri, 8 Mar 2019 09:08:38 -0800
Subject: [PATCH] IFNAMSIZ is defined in net/if.h

Fixes
/mnt/a/yoe/build/tmp/work/qemuriscv64-yoe-linux-musl/linux-atm/2.5.2-r0/recipe-sysroot/usr/include/linux/if_arp.h:121:16: error: 'IFNAMSIZ' undeclared here (not in a function)
|  char  arp_dev[IFNAMSIZ];
|                ^~~~~~~~
In file included from ../../../linux-atm-2.5.2/src/arpd/itf.c:17:
/mnt/a/yoe/build/tmp/work/qemuriscv64-yoe-linux-musl/linux-atm/2.5.2-r0/recipe-sysroot/usr/include/linux/if_arp.h:121:16: error: 'IFNAMSIZ' undeclared here (not in a function)
|  char  arp_dev[IFNAMSIZ];
|                ^~~~~~~~

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 src/arpd/arp.c | 1 +
 src/arpd/itf.c | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/arpd/arp.c b/src/arpd/arp.c
index 92d3787..ff1574c 100644
--- a/src/arpd/arp.c
+++ b/src/arpd/arp.c
@@ -17,6 +17,7 @@
 #include <netinet/in.h> /* for ntohs, etc. */
 #define _LINUX_NETDEVICE_H /* very crude hack for glibc2 */
 #include <linux/types.h>
+#include <net/if.h>
 #include <linux/if_arp.h>
 #include <linux/if_ether.h>
 #include <atm.h>
diff --git a/src/arpd/itf.c b/src/arpd/itf.c
index 92f0951..d285e3b 100644
--- a/src/arpd/itf.c
+++ b/src/arpd/itf.c
@@ -14,6 +14,7 @@
 #include <sys/socket.h>
 #define _LINUX_NETDEVICE_H /* glibc2 */
 #include <linux/types.h>
+#include <net/if.h>
 #include <linux/if_arp.h>
 
 #include "atmd.h"
-- 
2.21.0

